### PR TITLE
Paid to paid upgrade plan

### PR DESCRIPTION
**Pull request by jennysivapalan**
Enabled paid to paid upgrade using existing card details. We created PaymentSummary class that contains the current invoice and previous invoice which can be used to display a summary of the bill.

Note this feature is not enabled but backend code can now handle paid to paid upgrades. If this commit https://github.com/guardian/membership-frontend/commit/243a165c0708aa3f0cea41052ae35827c1904ac7 is reverted then it will all work. :tada: 

img_20150204_142303329

The outcome of pairing with @wpf500

### CODE REVIEW

File: frontend/app/services/SubscriptionService.scala [hunk 1]
```diff
@@ -123,4 +125,14 @@ class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String], val
   def createSubscription(memberId: MemberId, joinData: JoinForm, customerOpt: Option[Stripe.Customer]): Future[SubscribeResult] = {
     zuora.request(Subscribe(memberId, customerOpt, tierPlanRateIds(joinData.plan), joinData.name, joinData.deliveryAddress))
   }
+
+  def getPaymentSummary(memberId: MemberId): Future[PaymentSummary] = {
+    for {
+      subscription <- getSubscriptionStatus(memberId)
+      invoiceItems <- zuora.query[InvoiceItem](s"SubscriptionId='${subscription.current}'")
+    } yield {
+      val sortedInvoiceItems = sortInvoiceItems(invoiceItems)
```
Comment (rtyley) [inline frontend/app/services/SubscriptionService.scala]: The `sortInvoiceItems()` method has been extracted here for testing purposes, but obviously is a bit small. I'd be tempted to combine `PaymentSummaryTest` & the "sort invoice items by charge number ascending" test into a single test that does slightly more work, and change the method to be:

```
def paymentSummaryFor(items: Seq[InvoiceItem]): PaymentSummary = {
  val sortedInvoiceItems=items.sortBy(_.chargeNumber)
  PaymentSummary(sortedInvoiceItems.last, sortedInvoiceItems.dropRight(1))
}
```

Comment (jennysivapalan) [inline frontend/app/services/SubscriptionService.scala]: I made a change: https://github.com/guardian/membership-frontend/commit/8fa26222b3d8b6c26329697950520879ce6dcff7

Is this okay?

**Comment by wpf500** [comment]
@jamesoram this affects our current upgrade flows, you want to take a look?

**Comment by jamesoram** [comment]
When a friend tries to upgrade to a partner, when submitting the upgrade form there is no feedback, just a spinning wheel. The server returns a 200 and the tier actually gets changed though.

**Comment by feedmypixel** [comment]
> When a friend tries to upgrade to a partner, when submitting the upgrade form there is no feedback, just a spinning wheel. The server returns a 200 and the tier actually gets changed though. @feedmypixel Could this be related to #160 ?

Master is fine so i'd make sure this is up to date and let me know if there are any issues and I can look into it

**Comment by wpf500** [comment]
My bad

**Comment by feedmypixel** [comment]
screen shot 2015-02-05 at 18 03 48

as predicted...

**Comment by jamesoram** [comment]
:+1:

**Comment by jennysivapalan** [comment]
I used the test-users feature to Test the Friend -> Partner upgrade which worked fine. I also tested Partner to Patron which was disabled as expected. :boom:

**Comment by rtyley** [comment]
Awesome :+1:

**Pull request merged**